rmf_demo_plugins/rmf_plugins_common/src/readonly_common.cpp

```diff
@@ -0,0 +1,359 @@
+#include <rmf_plugins_common/readonly_common.hpp>
+#include <rmf_plugins_common/utils.hpp>
+
+using namespace rmf_readonly_common;
+
+using BuildingMap = building_map_msgs::msg::BuildingMap;
+using Level = building_map_msgs::msg::Level;
+using Graph = building_map_msgs::msg::Graph;
+using Location = rmf_fleet_msgs::msg::Location;
+using Path = std::vector<Location>;
+
+static double compute_yaw(const Eigen::Isometry3d& pose)
+{
+  auto quat = Eigen::Quaterniond(pose.linear());
+  // Taken from ignition math quaternion Euler()
+  double yaw = std::atan2(2 * (quat.x()*quat.y() + quat.w()*quat.z()),
+      (quat.w() * quat.w()) + (quat.x() * quat.x()) - (quat.y() * quat.y()) -
+      (quat.z() * quat.z()));
+  return yaw;
+}
+
+rclcpp::Logger ReadonlyCommon::logger()
+{
+  return rclcpp::get_logger("read_only_" + name);
+}
+
+void ReadonlyCommon::init(rclcpp::Node::SharedPtr node)
+{
+  _current_mode.mode = rmf_fleet_msgs::msg::RobotMode::MODE_MOVING;
+  _current_task_id = "demo";
+
+  ros_node = std::move(node);
+
+  _robot_state_pub =
+    ros_node->create_publisher<rmf_fleet_msgs::msg::RobotState>(
+    "/robot_state", 10);
+
+  auto qos_profile = rclcpp::QoS(10);
+  qos_profile.transient_local();
+  _building_map_sub = ros_node->create_subscription<BuildingMap>(
+    "/map",
+    qos_profile,
+    std::bind(&ReadonlyCommon::map_cb, this, std::placeholders::_1));
+}
+
+void ReadonlyCommon::on_update()
+{
+  _update_count++;
```

Comment:
Is this variable necessary? I can't find anywhere where it's used

Reply:
Nope, was a holdover from the old code. Will remove.